src/components/sbb-teaser/sbb-teaser.tsx

```diff
@@ -0,0 +1,112 @@
+import { Component, h, Prop, Watch, Element } from '@stencil/core';
+
+/**
+ * @slot image - Slot used to render the image
+ * @slot headline - Slot used to render the headline
+ * @slot description - Slot used to render the description
+ */
+
+@Component({
+  shadow: true,
+  styleUrls: {
+    default: 'styles/sbb-teaser.default.scss',
+    shared: 'styles/sbb-teaser.shared.scss',
+  },
+  tag: 'sbb-teaser',
+})
+
+/**
+ * Generalized Teaser - for displaying an image, headline and paragraph
+ */
+export class SbbTeaser {
+  /**
+   * The text which gets exposed to screen reader users. The text should
+   * reflect all the information
+   *
+   * Example text: Connection from X to Y, via Z, on date X.
+   * Ticket price starts at X.
+   */
+  @Prop() public accessibilityLabel!: string;
+
+  /**
+   * Check if accessibilityLabel is provided since it is a required prop,
+   * otherwise throw an error.
+   */
+  /* eslint-disable */
+  @Watch('accessibilityLabel')
+  validateAccessibilityLabel(newValue: string) {
+    const isBlank = typeof newValue !== 'string' || newValue === '';
+    if (isBlank) {
+      throw new Error('accessibilityLabel: required');
+    }
+  }
+  /* eslint-enable */
+
+  /**
+   * component attributes
+   * ----------------------------------------------------------------
+   */
+
+  /** The href value you want to link to */
+  @Prop() public hrefValue!: string;
+
+  /**
+   * Teaser variant -
+   * when this is true the text-content will be under the image
+   * otherwise it will be displayed next to the image.
+   */
+  @Prop() public isStacked: boolean;
+
+  @Element() private _element: HTMLElement;
+
+  public componentWillLoad(): void {
+    // Validate props
+    this.validateAccessibilityLabel(this.accessibilityLabel);
+  }
+
+  private _linkMutationObserver = new MutationObserver((mutations) => {
+    mutations?.forEach((mutation) => {
+      const element = mutation.target as HTMLElement;
+
+      if (element.nodeName === 'SBB-TITLE' && element.getAttribute('level') !== '5') {
+        element.setAttribute('level', '5');
+      }
+    });
+  });
```

Comment:
Can this new overriding behaviour be covered in a unit test?

Reply:
It seems like we only get the root before the connectedCallback() in jest. Do you know a way to test this behaviour?

Reply:
@dauriamarco Hello Marco, can you help here?